libs/pages/services/src/lib/feature/page-settings-preview-environments-feature/page-settings-preview-environments-feature.tsx

```diff
@@ -40,28 +30,37 @@ export function PageSettingsPreviewEnvironmentsFeature() {
       cloneEnvironmentDeploymentRules.auto_preview = data['auto_preview']
       cloneEnvironmentDeploymentRules.on_demand_preview = data['on_demand_preview']
 
-      await editEnvironmentDeploymentRule.mutate({
+      editEnvironmentDeploymentRule.mutate({
         environmentId,
         deploymentRuleId: environmentDeploymentRules?.id || '',
         data: cloneEnvironmentDeploymentRules,
       })
 
-      await applications?.forEach(async (application: ApplicationEntity) => {
-        if (application.id === Object.keys(data).find((key) => key === application.id)) {
-          const cloneApplication: ApplicationEntity = Object.assign({}, application as ApplicationEntity)
-          cloneApplication.auto_preview = data[application.id]
-
-          await dispatch(
-            editApplication({
-              applicationId: application.id,
-              data: cloneApplication,
-              serviceType: getServiceType(application),
-              silentToaster: true,
-              // eslint-disable-next-line @typescript-eslint/no-empty-function
-              toasterCallback: () => {},
-              queryClient,
+      services?.forEach(async (service: AnyService) => {
+        if (service.serviceType === 'DATABASE') return
+
+        if (service.id === Object.keys(data).find((key) => key === service.id)) {
+          const request = {
+            auto_preview: data[service.id],
+          }
+
+          const payload = match(service)
+            .with({ serviceType: 'APPLICATION' }, (s) => buildEditServicePayload({ service: s, request }))
+            .with({ serviceType: 'CONTAINER' }, (s) => buildEditServicePayload({ service: s, request }))
+            .with({ serviceType: 'JOB' }, (s) => buildEditServicePayload({ service: s, request }))
+            .with({ serviceType: 'HELM' }, (s) => buildEditServicePayload({ service: s, request }))
+            .otherwise(() => null)
+
+          if (!payload) return
```

Comment:
```suggestion
            .exhaustive()
```
as you have handle `DATABASE` type upfront it should be fine